src/Illuminate/Contracts/Auth/Guard.php

```diff
@@ -46,6 +48,13 @@ public function validate(array $credentials = []);
      * @return bool
      */
     public function hasUser();
+    
+    /**
+     * Compares the authenticated user with a given user.
+     *
+     * @return bool
+     */
+    public function userIs(AuthenticatableContract $user);
```

Comment:
The change to the contract makes this a breaking change if we are targeting 9.x

We could:

1. Target 10.x
2. Remove the contract change and just implement the method on the `GuardHelpers` trait to make it available on implementations.

Reply:
The Eloquent Model has an `is` method for this exact comparison:

````php
if (auth()->user()->is($post->user)) {
````